### ISSUE TITLE
get_details_permissions() working correctly?

### ISSUE THREAD

**Issue by RamonBeast** [androguard/androguard#639]
### Describe what you wanted to do

I'm getting odd results from calling ```get_details_permissions()``` as every permission is marked as unknown and 'normal'

### Describe what you expected

```
a, dx, d = AnalyzeAPK('package.apk')
a.get_details_permissions()
```
Given this list of permissions found:
```
['com.android.launcher.permission.INSTALL_SHORTCUT',
 'android.permission.READ_SETTINGS',
 'android.permission.WRITE_SETTINGS',
 'android.permission.ACCESS_NETWORK_STATE',
 'android.permission.WRITE_EXTERNAL_STORAGE',
 'android.permission.INTERNET',
 'android.permission.READ_PHONE_STATE',
 'android.permission.READ_SMS',
 'android.permission.SEND_SMS',
 'android.permission.RECEIVE_SMS',
 'android.permission.READ_CONTACTS']
```
I would expect to get appropriate levels, for instance ```READ_SMS``` should return ```dangerous```

### Describe what actually happened

This is the actual result:
```
{'android.permission.ACCESS_NETWORK_STATE': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.INTERNET': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.READ_CONTACTS': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.READ_PHONE_STATE': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.READ_SETTINGS': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.READ_SMS': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.RECEIVE_SMS': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.SEND_SMS': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.WRITE_EXTERNAL_STORAGE': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'android.permission.WRITE_SETTINGS': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference'],
 'com.android.launcher.permission.INSTALL_SHORTCUT': ['normal',
  'Unknown permission from android reference',
  'Unknown permission from android reference']}
```

### System Information

- Androguard Version: 3.3.4 (same happens for 3.2.1)
- Python Version: 3.7
- Operating Systems: Mac OS, Linux, Windows

**Comment by reox** [comment]
Can you share the apk?

Thanks in advance!

**Comment by RamonBeast** [comment]
The APK contains a malware, where would you prefer me to upload it?

The reason for this behavior is probably that the the permissions are accessed (https://github.com/androguard/androguard/tree/master/androguard/core/api_specific_resources/aosp_permissions) by using the ```targetSDK``` and apparently if not present (I have two samples, one with targetSDK 5 and another 26) the behavior is the one described which I believe is dangerous as permissions that are well defined appear as unknown with a protection level of "normal".

**Comment by reox** [comment]
Is it available on koodous? Or simply send me a sha256, i can usually find them ( if they are not too excotic).

Yeah, the json files need some update...

**Comment by RamonBeast** [comment]
Please find them here:

https://koodous.com/apks?search=0b3b5785f41f80bbb1fa70533cfa9acd24fd9262e9b868bcf1c6790fbf206898

https://koodous.com/apks?search=3ffff34a929611ae066f4156e407a288cc3718f735bc68c1f0deb21032ca34ad

What should be the appropriate behavior of ```get_details_permissions()``` in such cases?

**Comment by reox** [comment]
ok thank you!

I'm not sure what the default behaviour should be... If this was defined, you would not get this error :D 

I'll add some warning when the API level could not be loaded and try to load another version... probably thats the best I can do.

**Comment by RamonBeast** [comment]
Yes makes sense, the behavior might be to load the next higher version available or the highest available if the analysed targetSDK is above the highest one available (the 2 apk provided fall one in the first category and the other on the second).

Thanks for your prompt answers

**Comment by reox** [comment]
there is actually a default API setting.
it will change now to this:

```
$ androguard analyze 0b3b5785f41f80bbb1fa70533cfa9acd24fd9262e9b868bcf1c6790fbf206898                   
Please be patient, this might take a while.                                                                          
Found the provided file is of type 'APK'                                                                             
[WARNING ] androguard.default: API mapping for API level 5 was not found! Returning default, which is API level 16   
[INFO    ] androguard.analysis: End of creating cross references (XREF)                                              
[INFO    ] androguard.analysis: run time: 0min 00s                                                                   
Added file to session: SHA256::0b3b5785f41f80bbb1fa70533cfa9acd24fd9262e9b868bcf1c6790fbf206898                      
Loaded APK file...                                                                                                   
>>> a                                                                                                                
<androguard.core.bytecodes.apk.APK object at 0x0000000005768E10>                                                     
>>> d                                                                                                                
[<androguard.core.bytecodes.dvm.DalvikVMFormat object at 0x00000000101B3E10>]                                        
>>> dx                                                                                                               
<analysis.Analysis VMs: 1, Classes: 99, Strings: 106>                                                                
                                                                                                                     
Androguard version 3.3.4 started                                                                                     
In [1]: a.get_details_permissions()                                                                                  
Out[1]:                                                                                                              
{'android.permission.READ_CONTACTS': ['dangerous',                                                                   
  '@string/permlab_readContacts',                                                                                    
  '@string/permdesc_readContacts'],                                                                                  
 'android.permission.WRITE_EXTERNAL_STORAGE': ['dangerous',                                                          
  '@string/permlab_sdcardWrite',                                                                                     
  '@string/permdesc_sdcardWrite'],                                                                                   
 'android.permission.ACCESS_NETWORK_STATE': ['normal',                                                               
  '@string/permlab_accessNetworkState',                                                                              
  '@string/permdesc_accessNetworkState'],                                                                            
 'android.permission.INTERNET': ['dangerous',                                                                        
  '@string/permlab_createNetworkSockets',                                                                            
  '@string/permdesc_createNetworkSockets'],                                                                          
 'android.permission.READ_SETTINGS': ['normal',                                                                      
  'Unknown permission from android reference',                                                                       
  'Unknown permission from android reference'],                                                                      
 'android.permission.SEND_SMS': ['dangerous',                                                                        
  '@string/permlab_sendSms',                                                                                         
  '@string/permdesc_sendSms'],                                                                                       
 'com.android.launcher.permission.INSTALL_SHORTCUT': ['normal',                                                      
  '@string/permlab_install_shortcut',                                                                                
  '@string/permdesc_install_shortcut'],                                                                              
 'android.permission.RECEIVE_SMS': ['dangerous',                                                                     
  '@string/permlab_receiveSms',                                                                                      
  '@string/permdesc_receiveSms'],                                                                                    
 'android.permission.READ_PHONE_STATE': ['dangerous',                                                                
  '@string/permlab_readPhoneState',                                                                                  
  '@string/permdesc_readPhoneState'],                                                                                
 'android.permission.READ_SMS': ['dangerous',                                                                        
  '@string/permlab_readSms',                                                                                         
  '@string/permdesc_readSms'],                                                                                       
 'android.permission.WRITE_SETTINGS': ['dangerous',                                                                  
  '@string/permlab_writeSettings',                                                                                   
  '@string/permdesc_writeSettings']}           

```

Let's see if we can fix #529 as well.
